### PR TITLE
Add support for a label in product params

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Fixes https://github.com/web-platform-tests/wpt.fyi/issues/337 

Introduces a `ProductSpec` type, which holds and forwards any product-specific labels.
Note that this, incidentally, is a step towards `?product=chrome@sha`, but that's ignored for this PR.

## Review Information
Try things like ?product=chrome[stable]&product=chrome-69[experimental]&diff in the deployed env.

### CODE REVIEW

File: shared/params.go [hunk 1]
```diff
@@ -26,9 +26,45 @@ type TestRunFilter struct {
...
+// Products gets the slice of products specified in the ProductSpecs slice.
+func (p ProductSpecs) Products() []Product {
+	result := make([]Product, len(p))
+	for i, spec := range p {
+		result[i] = spec.Product
+	}
+	return result
+}
+
+func (p ProductSpec) String() string {
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply shared/params.go]: Interestingly, no; Must be a special case.

File: shared/params.go [hunk 2]
```diff
@@ -26,9 +26,45 @@ type TestRunFilter struct {
...
+		s += "["
+		for label := range p.Labels.Iter() {
+			s += label.(string) + ","
+		}
+		s = s[:len(s)-1]
+		s += "]"
+	}
+	return s
+}
+
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply shared/params.go]: Guessing you mean Len/Swap/Less. Same deal; no green underlines.

**Comment by lukebjerring** [comment]
@Hexcles - I'll leave you to merge the change once you've had a play (now that it's deployed).

**Comment by Hexcles** [comment]
@lukebjerring I'm seeing 500 from the underlying `/api/runs` call when trying the following queries:

https://product-labels-dot-wptdashboard-staging.appspot.com/results/?product=chrome-67[stable]
https://product-labels-dot-wptdashboard-staging.appspot.com/results/?product=chrome-69[experimental]

You should be able to find detailed errors in Stackdriver.

**Comment by lukebjerring** [comment]
That's just a missing index - I'll add that manually and then merge this.

**Pull request merged**